### PR TITLE
Consider properties defined in .mvn/maven.config

### PR DESCRIPTION
**Pull request by laeubi**
Currently lemminx does not consider properties that are defined in the .mvn/maven.config and therefore gives errors or otherwhise wrong results when parsing a project.

This adds support for detecting and reading these properties from the .mvn/maven.config

Fix https://github.com/eclipse/lemminx-maven/issues/549

### CODE REVIEW

File: lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/project/MavenProjectCache.java [hunk 1]
```diff
@@ -364,7 +374,8 @@ public Optional<MavenProject> getSnapshotProject(File file) {
 		 */
 		public MavenProject getSnapshotProject(DOMDocument document, String profileId, boolean resolve) {
 			// it would be nice to directly rebuild from Model instead of re-parsing text
-			ProjectBuildingRequest request = newProjectBuildingRequest(resolve);
+			ProjectBuildingRequest request = newProjectBuildingRequest(resolve, null); // TODO can we get the file from
```
Comment (mickaelistria) [inline lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/project/MavenProjectCache.java:377]: new File(document.getDocumentURI())

**Review by mickaelistria — CHANGES_REQUESTED** [review]

**Comment by mickaelistria** [comment]
Thanks, that's a good proposal. Just 1 addition suggested.

File: lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/project/MavenProjectCache.java (hunk 1 above)
Comment (laeubi) [reply lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/project/MavenProjectCache.java]: Done

**Comment by laeubi** [comment]
If it is good in general I would polish it a bit (e.g. caching the result of  config parsing maybe), possibly it would make sense to reload the file if save is performed and on typing always use a cached variant I assume that `SnapshotProject` would then always use a cached variant?

**Comment by mickaelistria** [comment]
> If it is good in general I would polish it a bit (e.g. caching the result of config parsing maybe), possibly it would make sense to reload the file if save is performed and on typing always use a cached variant

I think the gain here would be very small because loading projects involves much longer work, so I wouldn't invest in optimizing this particular bit. 

> use a cached variant I assume that SnapshotProject would then always use a cached variant?

I'm not sure I get the question correctly; but my undestanding is that if pom.xml is not the only input (maven.config is too), then the cached project, and it's consumers, needs to be removed from cache when the maven.config changes too. But as it's not a too frequent case to modify maven.config (vs pom.xml), I wouldn't make it a immediate requirement to immediately react on updates.

> Another issue is that a maven.config can provide profiles and different settings, this is currently not taken into account as well but better be solved in a different PR.

OK.

I let you look at what you want to change now, but IMO it's already in a good state to be merged. Please ping whenever you think it's ready to be merged.

**Comment by laeubi** [comment]
@mickaelistria if you are fine with it I'm fine with it as well, I tested it locally and it works as expected for me.

**Comment by mickaelistria** [comment]
Thank you for this very valuable addition!

**Pull request merged**